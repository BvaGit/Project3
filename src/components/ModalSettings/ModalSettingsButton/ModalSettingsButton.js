import React from 'react';

import Button from '../../Button';

const ModalSettingsButton = ({ id, textId, changeLogin, closeModal, isError }) => {
    const handleClick = () => {
        changeLogin();
        closeModal();
    }

    return (
        <Button 
            id={id}
            type="button"
            textId={textId}
            onClick={handleClick}
            disabled={!isError}
        />
    )
}

export default ModalSettingsButton;
