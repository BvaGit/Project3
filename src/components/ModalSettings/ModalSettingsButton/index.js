import { connect } from 'react-redux';

import ModalSettingsButton from './ModalSettingsButton';
import { ChangeCredits } from '../../../store/user/asyncActions';
import {getIsValidRegFields} from "../../../store/registration/selectors";

const mapStateToProps = (state) => ({
    isError: getIsValidRegFields(state),
});
const mapDispatchToProps = (dispatch) => ({
    changeLogin: () => dispatch(ChangeCredits())
})

export default connect(mapStateToProps, mapDispatchToProps)(ModalSettingsButton);
