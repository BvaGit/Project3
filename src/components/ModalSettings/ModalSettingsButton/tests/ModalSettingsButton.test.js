import React from 'react';
import { shallow, mount } from 'enzyme';
import ModalSettingsButton from '../ModalSettingsButton';
import { Provider } from 'react-redux';
import { IntlProvider } from "react-intl";


describe('Rooms', () => {
    
    it("Should match snapshot", () => {
        
        const component = shallowBiggerSmart(<ModalSettingsButton />)
        expect(component.html()).toMatchSnapshot()
    })
})

export const shallowBiggerSmart = (component, store) => {
    const core = store
        ? <Provider store={store}>{component}</Provider>
        : component;
    return shallow(<IntlProvider >{core}</IntlProvider>);
};