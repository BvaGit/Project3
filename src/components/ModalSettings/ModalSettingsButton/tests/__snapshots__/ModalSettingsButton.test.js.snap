// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rooms Should match snapshot 1`] = `"<button type=\\"button\\" class=\\"button\\" disabled=\\"\\">undefined</button>"`;
